crates/toolchain/edsl-deserialize/src/castf_extension.rs

```diff
@@ -0,0 +1,56 @@
+use derive_more::From;
+use openvm_circuit::{
+    arch::{SystemPort, VmExtension, VmInventory, VmInventoryBuilder, VmInventoryError},
+    circuit_derive::{Chip, ChipUsageGetter},
+};
+use openvm_circuit_derive::{AnyEnum, InstructionExecutor};
+use openvm_instructions::{UsizeOpcode, VmOpcode};
+use openvm_native_circuit::{
+    adapters::convert_adapter::ConvertAdapterChip, CastFChip, CastFCoreChip,
+};
+use openvm_native_compiler::CastfOpcode;
+use p3_field::PrimeField32;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Default, Serialize, Deserialize)]
+pub struct CastF;
+
+#[derive(ChipUsageGetter, Chip, InstructionExecutor, From, AnyEnum)]
+pub enum CastFExecutor<F: PrimeField32> {
```

Comment:
structure of this crate is weird: `CastF` should be in `openvm-native-circuit`. 

This crate should be renamed `openvm-native-transpiler`